crates/aide/src/operation.rs

```diff
@@ -370,3 +370,155 @@ pub fn add_parameters(
+        assert_eq!(operation, Operation::default());
+    }
+
+    #[test]
+    fn operation_io() {
+        #[derive(OperationIo)]
+        struct OperationInputOutput;
+        #[derive(OperationIo)]
+        #[aide(input, output)]
+        struct OperationInputOutput2;
+        #[derive(OperationIo)]
+        struct OperationInputOutputGeneric<T>(T);
+        #[derive(OperationIo)]
+        #[aide(input)]
+        struct OperationInput;
+        #[derive(OperationIo)]
+        #[aide(output)]
+        struct OperationOutput;
+
+        generate::in_context(|ctx| {
+            assert_default_input_impl::<OperationInputOutput>(ctx);
+            assert_default_output_impl::<OperationInputOutput>(ctx);
+
+            assert_default_input_impl::<OperationInputOutput2>(ctx);
+            assert_default_output_impl::<OperationInputOutput2>(ctx);
+
+            assert_default_input_impl::<OperationInputOutputGeneric<()>>(ctx);
+            assert_default_output_impl::<OperationInputOutputGeneric<()>>(ctx);
+
+            assert_default_input_impl::<OperationInput>(ctx);
+
+            assert_default_output_impl::<OperationOutput>(ctx);
+        });
+    }
+
+    #[test]
+    fn operation_io_with() {
+        struct ImplsOperationInput;
+        impl OperationInput for ImplsOperationInput {
+            fn operation_input(_ctx: &mut GenContext, operation: &mut Operation) {
+                // Changing a property of the operation so that we know this function was called
+                operation.deprecated = true;
+            }
+
+            fn inferred_early_responses(
+                _ctx: &mut GenContext,
+                _operation: &mut Operation,
+            ) -> Vec<(Option<StatusCode>, Response)> {
+                vec![(None, Response::default())]
+            }
+        }
+
+        struct ImplsOperationOutput;
+        impl OperationOutput for ImplsOperationOutput {
+            type Inner = ();
+
+            fn operation_response(
+                _ctx: &mut GenContext,
+                _operation: &mut Operation,
+            ) -> Option<Response> {
+                Some(Response::default())
+            }
+
+            fn inferred_responses(
+                _ctx: &mut GenContext,
+                _operation: &mut Operation,
+            ) -> Vec<(Option<StatusCode>, Response)> {
+                vec![(None, Response::default())]
+            }
+        }
+
+        #[derive(OperationIo)]
+        #[aide(
+            input_with = "ImplsOperationInput",
+            output_with = "ImplsOperationOutput"
+        )]
+        struct OperationIoWith;
+
+        generate::in_context(|ctx| {
+            let mut operation = Operation::default();
+
+            OperationIoWith::operation_input(ctx, &mut operation);
+            assert!(operation.deprecated);
+            assert_eq!(
+                OperationIoWith::inferred_early_responses(ctx, &mut operation),
+                vec![(None, Response::default())],
+            );
```

Comment:
This `assert_eq!` is currently failing because the `OperationInput` impl the derive macro creates for `input_with` does not override this function. But I'm not sure whether that is expected or a bug.
If it's a bug I could add the fix to this pr, otherwise I would change the test.